examples/shopping-cart/src/actions/index.js

```diff
@@ -1,50 +1,38 @@
 import shop from '../api/shop'
 import * as types from '../constants/ActionTypes'
 
-function receiveProducts(products) {
-  return {
-    type: types.RECEIVE_PRODUCTS,
-    products: products
-  }
-}
+const receiveProducts = products => ({
+  type: types.RECEIVE_PRODUCTS,
+  products: products
+})
 
-export function getAllProducts() {
-  return dispatch => {
-    shop.getProducts(products => {
-      dispatch(receiveProducts(products))
-    })
-  }
-}
+export const getAllProducts = () => dispatch => shop.getProducts(products => {
+  dispatch(receiveProducts(products))
+})
 
-function addToCartUnsafe(productId) {
-  return {
-    type: types.ADD_TO_CART,
-    productId
-  }
-}
+const addToCartUnsafe = productId => ({
+  type: types.ADD_TO_CART,
+  productId
+})
 
-export function addToCart(productId) {
-  return (dispatch, getState) => {
-    if (getState().products.byId[productId].inventory > 0) {
-      dispatch(addToCartUnsafe(productId))
-    }
+export const addToCart = productId => (dispatch, getState) => {
+  if (getState().products.byId[productId].inventory > 0) {
+    dispatch(addToCartUnsafe(productId))
   }
 }
 
-export function checkout(products) {
-  return (dispatch, getState) => {
-    const cart = getState().cart
+export const checkout = (products) => (dispatch, getState) => {
+  const cart = getState().cart
```

Comment:
If you're really going this far, you might as well do `const {cart} = getState()`.